src/main.ts

```diff
@@ -1,22 +1,49 @@
 import * as core from '@actions/core';
-import {install} from './cli.js';
+import path from 'path';
+import {fillEnv, install} from './cli.js';
 import {type Args, validateArgs} from './input.js';
+import {CLIError, run} from './run.js';
 
 
 const getArgs = (): Args => ({
     version: core.getInput('version', {
         required: true,
-    })
+    }),
+    access_key: core.getInput('access_key'),
+    secret_key: core.getInput('secret_key'),
+    default_organization_id: core.getInput('default_organization_id'),
+    default_project_id: core.getInput('default_project_id'),
+    args: core.getInput('args'),
 });
 
 export const main = async (): Promise<void> => {
     const args = getArgs();
 
     if (validateArgs(args)) {
-        return;
+        return undefined;
     }
 
-    await install(args.version);
+    const cliPath = await install(args.version);
+
+    if (args.args !== '') {
```

Comment:
En js je crois que si args.args est vide, tu peux faire 

```suggestion
    if (args.args) {
```

ou sinon si args est unknown

```suggestion
    if (typeof args.args === 'string' && args.args.length > 0) {
```

Enfin moyen de peaufiner je pense

Reply:
you're right